libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx

```diff
@@ -88,16 +88,23 @@ export function CreateUpdateVariableModal(props: CreateUpdateVariableModalProps)
     setTimeout(() => textarea?.focus(), 50)
   }
 
-  const availableScopes = computeAvailableScope(variable?.scope, false, scope, type === 'OVERRIDE') as Scope[]
+  // Compute available scopes and filter them - if it's a file, do not allow targeting a service scope
+  const availableScopes = (computeAvailableScope(variable?.scope, false, scope, type === 'OVERRIDE') as Scope[]).filter(
+    (s) => !_isFile || !['APPLICATION', 'CONTAINER', 'JOB', 'HELM'].includes(s)
```

Comment:
@RemiBonnet Could you explain why do you need `!['APPLICATION', 'CONTAINER', 'JOB', 'HELM'].includes(s)` ?

Reply:
@rmnbrd Yes I added it to remove the option inside `InputSelect` options, you can see this [here](https://github.com/Qovery/console/blob/539da5662144fd170f7a2811280089897cd6c89c/libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx#L448)